### DISCUSSION TITLE
Allow `<>` in match

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#4891] [Scripting]
```gdscript
match x:
    <10
        print("less than 10")
    _:
        print("else")
```

**Comment by dalexeev** [comment]
Related:

* #3733

I don't think this is a very useful feature. For more complex cases and for greater clarity, we still need to include the variable name:

```gdscript
match x:
    x < 0:
        print("Negative number")
    0 <= x <= 9:
        print("Digit")
    x >= 10:
        print("Multi-digit positive number")
```

Which is essentially the same as `if`-`elif`-`else`.

For numbers specifically, we could add a range pattern syntax, but I'm not sure if this will be used often:

```gdscript
match x:
    ..-1:
        print("Negative number")
    0..9:
        print("Digit")
    10..:
        print("Multi-digit positive number")
```

**Reply by dalexeev** [reply]
While it is not exactly what you suggested, [pattern guards](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#pattern-guards) allow you to use arbitrary expressions in `match` branches:

```gdscript
match x:
    _ when x < 10
        print("less than 10")
    _:
        print("else")
```

```gdscript
match x:
    _ when x < 0:
        print("Negative number")
    _ when 0 <= x and x <= 9:
        print("Digit")
    _ when x >= 10:
        print("Multi-digit positive number")
```

**Reply by DarcTellier** [reply]
This would be useful. Just went down a rabbit hole that led me here.
